test/unit/with-children.spec.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+/** @jsx h */
+/* eslint-env jest */
+
+import { mount } from '@skatejs/bore';
+import { h } from '@skatejs/val';
+import { define, withChildren, withUnique } from '../../src';
+
+// Very basic implementation so that the withChildren() mixin works.
+global.MutationObserver = class {
+  constructor(callback: Function) {
+    this.callback = callback;
+  }
+  observe(elem) {
+    const { callback } = this;
+    const oldAppendChild = elem.appendChild;
+    elem.appendChild = function(newChild) {
+      clearTimeout(this.timeout);
+      this.timeout = setTimeout(callback);
+      return oldAppendChild.call(this, newChild);
+    };
+  }
+};
+
+const Elem = define(
+  class extends withChildren(withUnique()) {
+    childrenChangedCallback() {
+      ++this.called;
+    }
+  }
+);
+
+describe('withChildren', () => {
```

Comment:
I wanted to add another test here to make sure observers won't double up if the node is removed and reattached, but then we'd be testing the mocked version.